godaddy/table_godaddy_dns_record.go

```diff
@@ -0,0 +1,184 @@
+			},
+			{
+				Name:        "service",
+				Description: "The type of the service.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "ttl",
+				Description: "The resource record cache time to live (TTL), in seconds.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("TTL"),
+			},
+			{
+				Name:        "weight",
+				Description: "Weighted resource record sets only: Among resource record sets that have the same combination of DNS name and type, a value that determines the proportion of DNS queries that Amazon Route 53 responds to using the current resource record set. Route 53 calculates the sum of the weights for the resource record sets that have the same combination of DNS name and type. Route 53 then responds to queries based on the ratio of a resource's weight to the total.",
+				Type:        proto.ColumnType_INT,
+			},
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: "Title of the resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+// The response data should include the domain name as it is the parent of the DNS record. However, the API response does not provide the domain name. Therefore, we need to create a custom struct to accommodate it.
+type DNSRecordInfo struct {
+	DomainName string
+	Data       string
+	Name       string
+	Port       int
+	Priority   int
+	Protocol   int
+	Service    int
+	TTL        int
+	Type       string
+	Weight     int
+}
+
+//// LIST FUNCTION
+
+func listDNSRecords(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	domain := h.Item.(*daddy.DomainDetail)
+
+	// Create Client
+	client, err := getClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("godaddy_dns_record.listDNSRecords", "client_error", err)
+		return nil, err
+	}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows.
+	// The API throws error (Page size may not exceed 500 records) if we are providing the max limit value grater than 500.
+	maxLimit := 500
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < int32(maxLimit) {
+			maxLimit = int(limit)
+		}
+	}
+
+	// Reduce the API calls for a given domain name in where clause.
+	domainName := domain.Domain
+	if d.EqualsQualString("domain_name") != "" && d.EqualsQualString("domain_name") != domainName {
+		return nil, nil
+	}
+
+	dnsType := ""
+	if d.EqualsQualString("type") != "" {
+		dnsType = d.EqualsQualString("type")
+	}
+
+	name := ""
+	if d.EqualsQualString("name") != "" {
+		name = d.EqualsQualString("name")
+	}
+
+	// To avoid errors, both the input parameters "name" and "DNS type" must be provided simultaneously in the where clause. Passing only one of them will result in an error being thrown by the API.
+	if name == "" || dnsType == "" {
+		name, dnsType = "", ""
+	}
```

Comment:
Is the logic correct? As per the above code, if either of the parameter is an empty string, then both the values get set to empty.

Reply:
@misraved Those were explored, and we didn't find any relevant data to consider. So we decided to stick to only current table sets.